src/core.js

```diff
@@ -3368,33 +3389,69 @@ export default function Core(rootElement, userSettings, rootInstanceSymbol = fal
     preventScrollingToCell = false;
   };
 
+  const getIndexToScroll = (indexMapper, visualIndex) => {
+    const firstColumnOnTheRight = indexMapper.getFirstNotHiddenIndex(visualIndex, 1);
+
+    if (firstColumnOnTheRight !== null) {
+      return firstColumnOnTheRight;
+    }
+
+    return indexMapper.getFirstNotHiddenIndex(visualIndex, -1); // Looking for a index on the left.
+  };
+
```

Comment:
I will think about it.

Reply:
Please look at my suggestion: https://github.com/handsontable/handsontable/pull/6547/commits/2d24a052d304185654735181889e9cee32dc2cdc. It's more generic, so someone else can use it (ie. for searrching at first on the left, then on the right). Anyway, this is draft, because I've found that:

`getFirstNotHiddenIndex(-1, 0)` return `-1`
`getFirstNotHiddenIndex(lastVisualIndex + 1)` return `0`
`getPhysicalFromVisualIndex(-1)` returns `undefined`
`getVisualFromPhysicalIndex(-1)` returns `null`
`getRenderableFromVisualIndex(-1)` returns `-1`

There are some inconsistencies. What's more, maybe it can be **not** a recursive function and it will speed up the search.

I have to think about fixing that and won't slow down the mapper. Maybe it should be considered with [this](https://github.com/handsontable/handsontable/pull/6547#discussion_r412918415) suggestion.

Reply:
The problem mentioned by me is fixed: https://github.com/handsontable/handsontable/commit/9b963a7c4572aab51ad253b593512c4f4fe62d2f#diff-2ab841eecb4411d5d9dbd452e431ee75R614-R626.